### PR TITLE
RSpec config update

### PR DESCRIPTION
**Pull request by orien**
Regenerate the `spec_helper` and `.rspec` files using the command:

```
rspec --init
```

This provides recommended options and prepares defaults for the next major release.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,17 +1,104 @@
...
+    # Use the documentation formatter for detailed output,
+    # unless a formatter has already been configured
+    # (e.g. via a command-line flag).
+    config.default_formatter = 'doc'
+  end
+
+  # Print the 10 slowest examples and example groups at the
+  # end of the spec run, to help surface which specs are running
+  # particularly slow.
+  # config.profile_examples = 10
```
Comment (orien) [inline spec/spec_helper.rb]: I've disabled this option as it's a little noisy and in my (loosely held) opinion less than helpful on this speedy test suite.

Comment (twe4ked) [reply spec/spec_helper.rb]: Agreed. This will be ignored unless people are explicitly looking to speed up the suite.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,17 +1,101 @@
...
+
+  # Many RSpec users commonly either run the entire suite or an individual
+  # file, and it's useful to allow more verbose output when running an
+  # individual spec file.
+  if config.files_to_run.one?
+    # Use the documentation formatter for detailed output,
+    # unless a formatter has already been configured
+    # (e.g. via a command-line flag).
+    config.default_formatter = "doc"
+  end
```
Comment (twe4ked) [inline spec/spec_helper.rb]: I personally find this unexpected. It's really easy to use `-fd` if you explicitly want doc format. No big deal though, just my opinion.

Comment (orien) [reply spec/spec_helper.rb]: Disabled in [`Don't use doc format when running one RSpec file`](https://github.com/envato/event_sourcery/pull/188/commits/e01006115770c1d7cf4e7545be483d70eb59bcb8)

**Comment by orien** [comment]
@twe4ked: Regarding the comments. For generated files, I tend to leave modifications to a minimum. I find this eases maintenance, especially when it comes time to upgrade to a later version.

**Pull request merged**